### PR TITLE
APIS-6001 Refactor existing ToU RI decline

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/thirdpartyapplication/scheduled/ResponsibleIndividualVerificationRemovalJob.scala [hunk 1]
```diff
@@ -72,33 +70,22 @@ class ResponsibleIndividualVerificationRemovalJob @Inject() (
...
-      case Standard(_, _, _, _, _, Some(importantSubmissionData)) => Some(importantSubmissionData.responsibleIndividual)
-      case _                                                      => None
-    }
-  }
-
-  private def getRequesterName(app: ApplicationData): Option[String] = {
-    app.state.requestedByName
-  }
+    (for {
+//      app            <- OptionT(applicationRepository.fetch(verificationDueForRemoval.applicationId))
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartyapplication/scheduled/ResponsibleIndividualVerificationRemovalJob.scala:78]: Commented code

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      isStandardNewJourneyApp(app),
+      isPendingResponsibleIndividualVerification(app),
+      isApplicationIdTheSame(app, riVerification),
+      isResponsibleIndividualDefined(app),
+      isRequesterEmailDefined(app),
+      isRequesterNameDefined(app)
+    ) { case _ => app }
+  }
+
+  // private def validateUpdate(app: ApplicationData, cmd: DeclineResponsibleIndividual, riVerification: ResponsibleIndividualUpdateVerification): ValidatedNec[String, ApplicationData] = {
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala:52]: Commented code

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        actor = CollaboratorActor(requesterEmail),
+        app.state.name,
+        State.TESTING,
+        requestingAdminName = requesterName,
+        requestingAdminEmail = requesterEmail
+      )
+    )
+  }
+
+  // private def asEventsUpdate(app: ApplicationData, cmd: DeclineResponsibleIndividual, riVerification: ResponsibleIndividualUpdateVerification): NonEmptyList[UpdateApplicationEvent] = {
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala:107]: Ditto

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  //   )  
+  // }
+
+  def process(app: ApplicationData, cmd: DeclineResponsibleIndividual): CommandHandler.Result = {
+    responsibleIndividualVerificationRepository.fetch(ResponsibleIndividualVerificationId(cmd.code)).map(maybeRIVerification => {
+      maybeRIVerification match {
+        case Some(riVerificationToU: ResponsibleIndividualToUVerification) => validateToU(app, cmd, riVerificationToU) map { _ =>
+          asEventsToU(app, cmd, riVerificationToU)
+        }
+        // case Some(riVerificationUpdate: ResponsibleIndividualUpdateVerification) => validateUpdate(app, cmd, riVerificationUpdate) map { _ =>
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala:132]: Ditto

File: conf/application.conf [hunk 5]
```diff
@@ -220,7 +220,7 @@ responsibleIndividualUpdateVerificationRemovalJob {
 responsibleIndividualVerificationSetDefaultTypeJob {
   initialDelay = 1m
   interval = 30d
-  enabled = true
+  enabled = false
```
Comment (AndySpaven) [inline conf/application.conf:223]: Does thus need corresponding config base change?

File: app/uk/gov/hmrc/thirdpartyapplication/scheduled/ResponsibleIndividualVerificationRemovalJob.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartyapplication/scheduled/ResponsibleIndividualVerificationRemovalJob.scala]: Removed

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala (hunk 3 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala]: Removed

File: conf/application.conf (hunk 5 above)
Comment (petekirby-ee) [reply conf/application.conf:223]: Removed

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala (hunk 4 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala]: Removed

File: app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartyapplication/services/commands/DeclineResponsibleIndividualCommandHandler.scala]: Removed

File: conf/application.conf (hunk 5 above)
Comment (petekirby-ee) [reply conf/application.conf:223]: No

**Review by AndySpaven — APPROVED** [review]

**Pull request merged**